catalog/transformer/catalog-transformer-csv-common/src/test/java/ddf/catalog/transformer/csv/common/CsvTransformerTest.java

```diff
@@ -175,6 +177,30 @@ public void writeSearchResultsToCsvWithAliasMap() throws CatalogTransformerExcep
     assertThat(scanner.hasNext(), is(false));
   }
 
+  @Test
+  public void filterEmptyValueAttributes() {
+    AttributeDescriptor attr9 = buildAttributeDescriptor("attribute9", BasicTypes.STRING_TYPE);
+    Set<AttributeDescriptor> attributeDescriptors = new HashSet<>(ATTRIBUTE_DESCRIPTOR_LIST);
+    attributeDescriptors.add(attr9);
+    MetacardType metacardType = new MetacardTypeImpl("", attributeDescriptors);
+    Metacard metacard = new MetacardImpl(metacardType);
+    for (Attribute a : metacardDataMap.values()) {
+      metacard.setAttribute(a);
+    }
+    metacard.setAttribute(new AttributeImpl("attribute9", ""));
+    List<Metacard> metacards = metacardList.stream().collect(Collectors.toList());
+    metacards.add(metacard);
+    Set<AttributeDescriptor> nonEmptyValues = CsvTransformer.getNonEmptyValueAttributes(metacards);
+
+    assertThat(nonEmptyValues, hasSize(6));
+    final Optional<AttributeDescriptor> attributeDescriptorOptional =
+        nonEmptyValues.stream().findFirst();
+    assertThat(attributeDescriptorOptional.isPresent(), is(true));
+
+    final AttributeDescriptor attributeDescriptor = attributeDescriptorOptional.get();
+    assertThat(attributeDescriptor, notNullValue());
```

Comment:
This doesn't verify what the test needs to verify. It should check that `nonEmptyValues` _doesn't_ contain any attributes which have no values in the supplied metacards.

Reply:
Good point. Carry over from copy / paste - updated the assertion.